spec/system/admin/reports/orders_and_distributors_spec.rb

```diff
@@ -0,0 +1,179 @@
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+      let(:line_item4) {
+        [completed_at.to_s, order.id, "John Doe", order.email, "123-456-7890", "Herndon",
+         "ABC", Spree::Product.first.name.to_s, "1g", "1", "none", "10.0", "none", "Check",
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+      let(:line_item5) {
+        [completed_at.to_s, order.id, "John Doe", order.email, "123-456-7890", "Herndon",
+         "ABC", Spree::Product.first.name.to_s, "1g", "1", "none", "10.0", "none", "Check",
+         "By Bike", "10 Lovely Street", "Herndon", "20170", "UPS Ground", "none"].join(" ")
+      }
+
+      before do
+        login_as(distributor.owner)
+        visit admin_reports_path
+        click_link "Orders And Distributors"
+        run_report
+      end
+
+      it "generates the report" do
+        rows = find("table.report__table").all("thead tr")
+        table_headers = rows.map { |r| r.all("th").map { |c| c.text.strip } }
+
+        expect(table_headers).to eq([header])
+
+        expect(all('table.report__table tbody tr').count).to eq(
+          Spree::LineItem.where(
+            order_id: order.id # Total rows should equal nr. of line items, per order
+          ).count
+        )
+
+        # displays only orders from the hub it is managing
+        expect(page).to have_content(distributor.name, count: 6)
+
+        # only sees line items from orders it manages
+        expect(page).not_to have_content(distributor2.name)
+
+        # displayes table contents correctly, per line item
+        table = page.find("table.report__table tbody")
+        expect(table).to have_content(line_item1)
+        expect(table).to have_content(line_item2)
+        expect(table).to have_content(line_item3)
+        expect(table).to have_content(line_item4)
+        expect(table).to have_content(line_item5)
+      end
+
+      describe "downloading the report" do
+        shared_examples "reports generated as" do |output_type, extension|
+          context output_type.to_s do
+            it "downloads the #{output_type} file" do
+              select output_type, from: "report_format"
+
+              expect { generate_report }.to change { downloaded_filenames.length }.from(0).to(1)
+
+              expect(downloaded_filename).to match(/.*\.#{extension}/)
+
+              downloaded_file_txt = load_file_txt(extension, downloaded_filename)
+
+              expect(downloaded_file_txt).to have_text header.join(" ")
+              expect(downloaded_file_txt).to have_text(
+                "By Bike 10 Lovely Street Herndon 20170 UPS Ground", count: 5
+              )
+            end
+          end
+        end
+
+        it_behaves_like "reports generated as", "CSV", "csv"
+        it_behaves_like "reports generated as", "Spreadsheet", "xlsx"
```

Comment:
Could it be that, depending on the version of Chrome, the PDF is opened in the browser instead of downloaded? That would explain the error, seeing a timeout because the file is never saved, only displayed.

Reply:
> PDF is opened in the browser instead of downloaded

I'm positive, that this is indeed the case - I've verified this in headless=false mode. But I have not found a way to download the file...